### PR TITLE
Add unicloud

### PR DESCRIPTION
**Pull request by MAfarrag**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MAfarrag** [comment]
@conda-forge/python @conda-forge-admin all tests have passed, can you check please.

File: recipes/unicloud/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: ddac4a0573c2b86c519ae02fcc8ed1d0390c7619d277756f1db26d66c522b1d6
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (ocefpaf) [inline recipes/unicloud/meta.yaml:19]: Is this correct? If so, this package cannot be noarch.

Comment (MAfarrag) [reply recipes/unicloud/meta.yaml]: removed

File: recipes/unicloud/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{% set name = "unicloud" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Serapieum-of-alex/unicloud/archive/{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/unicloud/meta.yaml:9]: Why not use the PyPI sdist? It is preferred b/c the hash won't change if the release notes change like on GitHub.

Comment (MAfarrag) [reply recipes/unicloud/meta.yaml:9]: I prefer to make both the pypi package and the conda-forge package pull the files from github.

Comment (ocefpaf) [reply recipes/unicloud/meta.yaml:9]: Note that, for Python packages, our automated tools can take advantage of PyPI for a better package. Things like the noarch + compiler would not happen when creating a recipe with grayskull from PyPI.

File: recipes/unicloud/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  sha256: ddac4a0573c2b86c519ae02fcc8ed1d0390c7619d277756f1db26d66c522b1d6
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/unicloud/meta.yaml:14]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

**Pull request merged**